File: Frontend/src/app/models/create-snapshot-dto.ts
import {CreateParallelizeOperationDto} from './create-parallelize-operation-dto';
import {CreateEliminateOperationDto} from './create-eliminate-operation-dto';

export interface CreateSnapshotDto {
  snapshotName: string;
  parallelizeOperation?: CreateParallelizeOperationDto;
  eliminateOperation?: CreateEliminateOperationDto;
}

File: Backend/Uniterminator/Features/Snapshots/DTOs/CreateSnapshotDto.cs
using System.Diagnostics.CodeAnalysis;

namespace Uniterminator.DTOs;

[SuppressMessage("ReSharper", "UnusedMember.Global")]
[SuppressMessage("ReSharper", "UnusedAutoPropertyAccessor.Global")]
public abstract record CreateSnapshotDto
{
    public DateTime SnapshotTime { get; init; }
    public CreateParallelizeOperationDto? ParallelizeOperation { get; init; }
    public CreateEliminateOperationDto? EliminateOperation { get; init; }
}